internal/resolution/variablesources/bundle_uniqueness_test.go

```diff
@@ -17,6 +21,97 @@ import (
 	"github.com/operator-framework/operator-controller/internal/resolution/variablesources"
 )
 
+func TestMakeBundleUniquenessVariables(t *testing.T) {
+	const fakeCatalogName = "fake-catalog"
+	channel := catalogmetadata.Channel{Channel: declcfg.Channel{Name: "stable"}}
+	bundleSet := map[string]*catalogmetadata.Bundle{
+		"test-package.v1.0.0": {
+			Bundle: declcfg.Bundle{
+				Name:    "test-package.v1.0.0",
+				Package: "test-package",
+				Properties: []property.Property{
+					{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "1.0.0"}`)},
+					{Type: property.TypePackageRequired, Value: json.RawMessage(`{"packageName": "some-package", "versionRange": ">=1.0.0 <2.0.0"}`)},
+				},
+			},
+			CatalogName: fakeCatalogName,
+			InChannels:  []*catalogmetadata.Channel{&channel},
+		},
+		"test-package.v1.0.1": {
+			Bundle: declcfg.Bundle{
+				Name:    "test-package.v1.0.1",
+				Package: "test-package",
+				Properties: []property.Property{
+					{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "1.0.1"}`)},
+					{Type: property.TypePackageRequired, Value: json.RawMessage(`{"packageName": "some-package", "versionRange": ">=1.0.0 <2.0.0"}`)},
+				},
+			},
+			CatalogName: fakeCatalogName,
+			InChannels:  []*catalogmetadata.Channel{&channel},
+		},
+
+		"some-package.v1.0.0": {
+			Bundle: declcfg.Bundle{
+				Name:    "some-package.v1.0.0",
+				Package: "some-package",
+				Properties: []property.Property{
+					{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "some-package", "version": "1.0.0"}`)},
+				},
+			},
+			CatalogName: fakeCatalogName,
+			InChannels:  []*catalogmetadata.Channel{&channel},
+		},
+	}
+
+	// Input into the testable function must include more than one bundle
+	// from the same package to ensure that the function
+	// enforces uniqueness correctly.
+	// We also need at least one bundle variable to have a dependency
+	// on another package. This will help to make sure that the function
+	// also creates uniqueness variables for dependencies.
+	bundleVariables := []*olmvariables.BundleVariable{
+		olmvariables.NewBundleVariable(
+			bundleSet["test-package.v1.0.0"],
+			[]*catalogmetadata.Bundle{
+				bundleSet["some-package.v1.0.0"],
+			},
+		),
+		olmvariables.NewBundleVariable(
+			bundleSet["test-package.v1.0.1"],
+			[]*catalogmetadata.Bundle{
+				bundleSet["some-package.v1.0.0"],
+			},
+		),
+	}
+
+	variables := variablesources.MakeBundleUniquenessVariables(bundleVariables)
+	require.Len(t, variables, 2)
```

Comment:
Is it possible to simply create a slice of struct literals here and use `cmp.Diff()` for testing that the output is exactly as expected?

Reply:
I think go-cmp ignores unexported fields by default. I'll see if I can make it work with `AllowUnexported`.

Reply:
Updated with go-cmp & `AllowUnexported` option. Looks cleaner.